Client/Ecosia/UI/WhatsNew/DataProvider/WhatsNewLocalDataProvider.swift

```diff
@@ -19,15 +19,32 @@ final class WhatsNewLocalDataProvider: WhatsNewDataProvider {
         Version(versionProvider.version)!
     }
     
-    /// This value can be used to determine if the user should be presented with the What's New page.
-    ///
+    /// A computed property to determine whether the "What's New" page should be displayed.
     /// - Returns: `true` if the What's New page should be shown; otherwise, `false`.
     var shouldShowWhatsNewPage: Bool {
-        return EcosiaInstallType.get() == .upgrade
+        
+        // Check if we are in the upgrade scenario
+        guard EcosiaInstallType.get() == .upgrade else {
```

Comment:
I understand your point here 👍 . The purpose of utilising the `User.shared.firstTime` solely served the purpose for additionally cover the scenario in case we'd need to show the What's New page upon upgrading to a version of the app that didn't have the logic of `EcosiaInstallType` at all (hence treating the app install as `.fresh`). What we want, in this scenario, is to set the `EcosiaInstallType` as `.upgrade` so that the UX will coherently behave.

Reply:
Yeah, totally understand the reasoning for this workaround. While going through my [other answer](https://github.com/ecosia/ios-browser/pull/564#discussion_r1345536839), also realised the two are not completely interchangeable solely due to the following order of execution:
1. `EcosiaInstallType.evaluateCurrentEcosiaInstallTypeWithVersionProvider`
2. `User.shared.firstTime = false`
3. `WhatsNewLocalDataProvider.shouldShowWhatsNewPage`

Like I mentioned on the other, it still feels error prone since it is not visible that this order matters for it to work and this might change in the future. But I also have to say I do not have a better suggestion for a different approach that is not overly complex, so don't object with going for it.

Reply:
> since it is not visible that this order matters for it to work
I 💯 agree with your point here.
I'm also open to complex solutions as long as they will eventually provide us with a solid approach and a decent value added time allocation wise. @lucaschifino up to talk on it!

Reply:
Me too! I think there is no reason to block this release because of it, but it is worth talking and seeing if we can come up with a better solution for another iteration 👍